### PR TITLE
[java] Update rule UseIOStreamsWithApacheCommonsFileItem

### PR DESCRIPTION
**Pull request by adangel**
Part of #2701 

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by travis)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to pmd/7.0.x:<br />
This changeset changes 0 violations,<br />
introduces 7 new violations, 376 new errors and 0 new configuration errors,<br />
removes 134 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/2cfa228861874681941d36bbb098fcc5/diff1/index.html">Full report</a></td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 164 new violations, 377 new errors and 0 new configuration errors,<br />
removes 285938 violations, 10 errors and 2 configuration errors.<br />
<a href="https://chunk.io/pmd/2cfa228861874681941d36bbb098fcc5/diff2/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by oowekyala** [comment]
The diff report has many new errors, I'll look into it tomorrow

File: pmd-java/src/main/resources/category/java/performance.xml [hunk 1]
```diff
@@ -1005,23 +1005,10 @@ and buffering.
...
-    ) or
-    starts-with(@Image, concat(
-        ancestor::ClassOrInterfaceBody/ClassOrInterfaceBodyDeclaration/FieldDeclaration/Type/ReferenceType/ClassOrInterfaceType[pmd-java:typeIs('org.apache.commons.fileupload.FileItem')]/../../..//VariableDeclaratorId/@Name,
-        '.')
-    )
-]
-]]>
+//MethodCall
+    [@MethodName = 'get' or @MethodName = 'getString']
+    [VariableAccess[pmd-java:typeIs('org.apache.commons.fileupload.FileItem')]]
```
Comment (oowekyala) [inline pmd-java/src/main/resources/category/java/performance.xml:1010]: Does this need to be a VariableAccess? it could be any expression I think

Comment (oowekyala) [reply pmd-java/src/main/resources/category/java/performance.xml:1010]: `typeIs` doesn't throw anymore when a wrong node is given so you can just do `*[pmd-java:typeIs(...)]`

Comment (adangel) [reply pmd-java/src/main/resources/category/java/performance.xml:1010]: That would catch stuff like

```java
private FileItem getItem() { .... }

private void bar() {
    String result = getItem().getString();
}
```

I can add this....

**Comment by adangel** [comment]
I think, this is ready now. Only one additional violation in Spring: https://github.com/spring-projects/spring-framework/blob/v5.0.6.RELEASE/spring-web/src/main/java/org/springframework/web/multipart/commons/CommonsMultipartFile.java#L139
But this is one correct.

**Review by oowekyala — APPROVED** [review]

**Pull request merged**